[BUG] test_iceberg_fallback_not_unsafe_row failed in databricks 10.4 runtime

**Describe the bug**
This was found in my PR to unify test mode
```
[2022-07-19T08:12:11.288Z] FAILED ../../src/main/python/iceberg_test.py::test_iceberg_fallback_not_unsafe_row[ALLOW_NON_GPU(BatchScanExec)]
[2022-07-19T08:12:11.288Z] ERROR ../../src/main/python/iceberg_test.py::test_iceberg_fallback_not_unsafe_row[ALLOW_NON_GPU(BatchScanExec)]
```

failed `: java.lang.IncompatibleClassChangeError: Implementing class`, pytest log,
```
[2022-07-19T08:12:11.285Z] =================================== FAILURES ===================================

[2022-07-19T08:12:11.285Z] _____________________ test_iceberg_fallback_not_unsafe_row _____________________

[2022-07-19T08:12:11.285Z] 

[2022-07-19T08:12:11.285Z] spark_tmp_table_factory = <conftest.TmpTableFactory object at 0x7f07bb07b490>

[2022-07-19T08:12:11.285Z] 

[2022-07-19T08:12:11.285Z]     @allow_non_gpu('BatchScanExec')

[2022-07-19T08:12:11.285Z]     @iceberg

[2022-07-19T08:12:11.285Z]     def test_iceberg_fallback_not_unsafe_row(spark_tmp_table_factory):

[2022-07-19T08:12:11.285Z]         table = spark_tmp_table_factory.get()

[2022-07-19T08:12:11.285Z]         def setup_iceberg_table(spark):

[2022-07-19T08:12:11.285Z]             spark.sql("CREATE TABLE {} (id BIGINT, data STRING) USING ICEBERG".format(table))

[2022-07-19T08:12:11.285Z]             spark.sql("INSERT INTO {} VALUES (1, 'a'), (2, 'b'), (3, 'c')".format(table))

[2022-07-19T08:12:11.285Z] >       with_cpu_session(setup_iceberg_table)

[2022-07-19T08:12:11.285Z] 

[2022-07-19T08:12:11.285Z] ../../src/main/python/iceberg_test.py:29: 

[2022-07-19T08:12:11.285Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2022-07-19T08:12:11.285Z] ../../src/main/python/spark_session.py:114: in with_cpu_session

[2022-07-19T08:12:11.285Z]     return with_spark_session(func, conf=copy)

[2022-07-19T08:12:11.285Z] ../../src/main/python/spark_session.py:98: in with_spark_session

[2022-07-19T08:12:11.285Z]     ret = func(_spark)

[2022-07-19T08:12:11.285Z] ../../src/main/python/iceberg_test.py:27: in setup_iceberg_table

[2022-07-19T08:12:11.285Z]     spark.sql("CREATE TABLE {} (id BIGINT, data STRING) USING ICEBERG".format(table))

[2022-07-19T08:12:11.285Z] /databricks/spark/python/pyspark/sql/session.py:777: in sql

[2022-07-19T08:12:11.285Z]     return DataFrame(self._jsparkSession.sql(sqlQuery), self._wrapped)

[2022-07-19T08:12:11.285Z] /databricks/spark/python/lib/py4j-0.10.9.1-src.zip/py4j/java_gateway.py:1304: in __call__

[2022-07-19T08:12:11.285Z]     return_value = get_return_value(

[2022-07-19T08:12:11.285Z] /databricks/spark/python/pyspark/sql/utils.py:117: in deco

[2022-07-19T08:12:11.285Z]     return f(*a, **kw)

[2022-07-19T08:12:11.285Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2022-07-19T08:12:11.285Z] 

[2022-07-19T08:12:11.285Z] answer = 'xro378'

[2022-07-19T08:12:11.285Z] gateway_client = <py4j.clientserver.JavaClient object at 0x7f07d9551d30>

[2022-07-19T08:12:11.285Z] target_id = 'o65', name = 'sql'

[2022-07-19T08:12:11.285Z] 

[2022-07-19T08:12:11.285Z]     def get_return_value(answer, gateway_client, target_id=None, name=None):

[2022-07-19T08:12:11.285Z]         """Converts an answer received from the Java gateway into a Python object.

[2022-07-19T08:12:11.285Z]     

[2022-07-19T08:12:11.285Z]         For example, string representation of integers are converted to Python

[2022-07-19T08:12:11.285Z]         integer, string representation of objects are converted to JavaObject

[2022-07-19T08:12:11.285Z]         instances, etc.

[2022-07-19T08:12:11.285Z]     

[2022-07-19T08:12:11.285Z]         :param answer: the string returned by the Java gateway

[2022-07-19T08:12:11.285Z]         :param gateway_client: the gateway client used to communicate with the Java

[2022-07-19T08:12:11.285Z]             Gateway. Only necessary if the answer is a reference (e.g., object,

[2022-07-19T08:12:11.285Z]             list, map)

[2022-07-19T08:12:11.285Z]         :param target_id: the name of the object from which the answer comes from

[2022-07-19T08:12:11.285Z]             (e.g., *object1* in `object1.hello()`). Optional.

[2022-07-19T08:12:11.285Z]         :param name: the name of the member from which the answer comes from

[2022-07-19T08:12:11.285Z]             (e.g., *hello* in `object1.hello()`). Optional.

[2022-07-19T08:12:11.286Z]         """

[2022-07-19T08:12:11.286Z]         if is_error(answer)[0]:

[2022-07-19T08:12:11.286Z]             if len(answer) > 1:

[2022-07-19T08:12:11.286Z]                 type = answer[1]

[2022-07-19T08:12:11.286Z]                 value = OUTPUT_CONVERTER[type](answer[2:], gateway_client)

[2022-07-19T08:12:11.286Z]                 if answer[1] == REFERENCE_TYPE:

[2022-07-19T08:12:11.286Z] >                   raise Py4JJavaError(

[2022-07-19T08:12:11.286Z]                         "An error occurred while calling {0}{1}{2}.\n".

[2022-07-19T08:12:11.286Z]                         format(target_id, ".", name), value)

[2022-07-19T08:12:11.286Z] E                   py4j.protocol.Py4JJavaError: An error occurred while calling o65.sql.

[2022-07-19T08:12:11.286Z] E                   : java.lang.IncompatibleClassChangeError: Implementing class

[2022-07-19T08:12:11.286Z] E                   	at java.lang.ClassLoader.defineClass1(Native Method)

[2022-07-19T08:12:11.286Z] E                   	at java.lang.ClassLoader.defineClass(ClassLoader.java:757)

[2022-07-19T08:12:11.286Z] E                   	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)

[2022-07-19T08:12:11.286Z] E                   	at java.net.URLClassLoader.defineClass(URLClassLoader.java:468)

[2022-07-19T08:12:11.286Z] E                   	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)

[2022-07-19T08:12:11.286Z] E                   	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)

[2022-07-19T08:12:11.286Z] E                   	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)

[2022-07-19T08:12:11.286Z] E                   	at java.security.AccessController.doPrivileged(Native Method)

[2022-07-19T08:12:11.286Z] E                   	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)

[2022-07-19T08:12:11.286Z] E                   	at java.lang.ClassLoader.loadClass(ClassLoader.java:419)

[2022-07-19T08:12:11.286Z] E                   	at java.lang.ClassLoader.loadClass(ClassLoader.java:352)

[2022-07-19T08:12:11.286Z] E                   	at org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions.$anonfun$apply$9(IcebergSparkSessionExtensions.scala:56)

[2022-07-19T08:12:11.286Z] E                   	at org.apache.spark.sql.SparkSessionExtensions.$anonfun$buildResolutionRules$1(SparkSessionExtensions.scala:152)

[2022-07-19T08:12:11.286Z] E                   	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)

[2022-07-19T08:12:11.286Z] E                   	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)

[2022-07-19T08:12:11.286Z] E                   	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)

[2022-07-19T08:12:11.286Z] E                   	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)

[2022-07-19T08:12:11.286Z] E                   	at scala.collection.TraversableLike.map(TraversableLike.scala:286)

[2022-07-19T08:12:11.286Z] E                   	at scala.collection.TraversableLike.map$(TraversableLike.scala:279)

[2022-07-19T08:12:11.286Z] E                   	at scala.collection.AbstractTraversable.map(Traversable.scala:108)

[2022-07-19T08:12:11.286Z] E                   	at org.apache.spark.sql.SparkSessionExtensions.buildResolutionRules(SparkSessionExtensions.scala:152)

[2022-07-19T08:12:11.286Z] E                   	at com.databricks.sql.DatabricksEdge.customResolutionRules(DatabricksEdge.scala:85)

[2022-07-19T08:12:11.286Z] E                   	at com.databricks.sql.DatabricksEdge.customResolutionRules$(DatabricksEdge.scala:72)

[2022-07-19T08:12:11.286Z] E                   	at org.apache.spark.sql.hive.HiveSessionStateBuilder.customResolutionRules(HiveSessionStateBuilder.scala:53)

[2022-07-19T08:12:11.286Z] E                   	at org.apache.spark.sql.hive.HiveSessionStateBuilder$$anon$1.<init>(HiveSessionStateBuilder.scala:114)

[2022-07-19T08:12:11.286Z] E                   	at org.apache.spark.sql.hive.HiveSessionStateBuilder.analyzer(HiveSessionStateBuilder.scala:97)

[2022-07-19T08:12:11.286Z] E                   	at org.apache.spark.sql.internal.BaseSessionStateBuilder.$anonfun$build$7(BaseSessionStateBuilder.scala:408)

[2022-07-19T08:12:11.286Z] E                   	at org.apache.spark.sql.internal.SessionState.analyzer$lzycompute(SessionState.scala:104)

[2022-07-19T08:12:11.286Z] E                   	at org.apache.spark.sql.internal.SessionState.analyzer(SessionState.scala:104)

[2022-07-19T08:12:11.286Z] E                   	at org.apache.spark.sql.execution.QueryExecution.$anonfun$analyzed$1(QueryExecution.scala:128)

[2022-07-19T08:12:11.286Z] E                   	at com.databricks.spark.util.FrameProfiler$.record(FrameProfiler.scala:80)

[2022-07-19T08:12:11.286Z] E                   	at org.apache.spark.sql.catalyst.QueryPlanningTracker.measurePhase(QueryPlanningTracker.scala:151)

[2022-07-19T08:12:11.287Z] E                   	at org.apache.spark.sql.execution.QueryExecution.$anonfun$executePhase$1(QueryExecution.scala:265)

[2022-07-19T08:12:11.287Z] E                   	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:968)

[2022-07-19T08:12:11.287Z] E                   	at org.apache.spark.sql.execution.QueryExecution.executePhase(QueryExecution.scala:265)

[2022-07-19T08:12:11.287Z] E                   	at org.apache.spark.sql.execution.QueryExecution.analyzed$lzycompute(QueryExecution.scala:129)

[2022-07-19T08:12:11.287Z] E                   	at org.apache.spark.sql.execution.QueryExecution.analyzed(QueryExecution.scala:126)

[2022-07-19T08:12:11.287Z] E                   	at org.apache.spark.sql.execution.QueryExecution.assertAnalyzed(QueryExecution.scala:118)

[2022-07-19T08:12:11.287Z] E                   	at org.apache.spark.sql.Dataset$.$anonfun$ofRows$2(Dataset.scala:103)

[2022-07-19T08:12:11.287Z] E                   	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:968)

[2022-07-19T08:12:11.287Z] E                   	at org.apache.spark.sql.Dataset$.ofRows(Dataset.scala:101)

[2022-07-19T08:12:11.287Z] E                   	at org.apache.spark.sql.SparkSession.$anonfun$sql$1(SparkSession.scala:803)

[2022-07-19T08:12:11.287Z] E                   	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:968)

[2022-07-19T08:12:11.287Z] E                   	at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:798)

[2022-07-19T08:12:11.287Z] E                   	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

[2022-07-19T08:12:11.287Z] E                   	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)

[2022-07-19T08:12:11.287Z] E                   	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)

[2022-07-19T08:12:11.287Z] E                   	at java.lang.reflect.Method.invoke(Method.java:498)

[2022-07-19T08:12:11.287Z] E                   	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)

[2022-07-19T08:12:11.287Z] E                   	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:380)

[2022-07-19T08:12:11.287Z] E                   	at py4j.Gateway.invoke(Gateway.java:295)

[2022-07-19T08:12:11.287Z] E                   	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)

[2022-07-19T08:12:11.287Z] E                   	at py4j.commands.CallCommand.execute(CallCommand.java:79)

[2022-07-19T08:12:11.287Z] E                   	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:195)

[2022-07-19T08:12:11.287Z] E                   	at py4j.ClientServerConnection.run(ClientServerConnection.java:115)

[2022-07-19T08:12:11.287Z] E                   	at java.lang.Thread.run(Thread.java:748)

[2022-07-19T08:12:11.287Z] 

[2022-07-19T08:12:11.287Z] /databricks/spark/python/lib/py4j-0.10.9.1-src.zip/py4j/protocol.py:326: Py4JJavaError
```



9.1 runtime works fine.

--- a/integration_tests/src/main/python/conftest.py
+++ b/integration_tests/src/main/python/conftest.py
@@ -183,6 +183,8 @@ def pytest_runtest_setup(item):
     if item.get_closest_marker('iceberg'):
         if not item.config.getoption('iceberg'):
             pytest.skip('Iceberg tests not configured to run')
+        elif is_databricks_runtime():
+            pytest.skip('Iceberg tests skipped on Databricks')
 
 def pytest_configure(config):
     global _runtime_env

